src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java

```diff
@@ -3420,9 +3420,47 @@
     public static final int ELLIPSIS = GeneratedJavaTokenTypes.ELLIPSIS;
 
     /**
-     * The {@code &} symbol when used in a generic upper or lower bounds constrain
-     * e.g&#46; <code>Comparable&lt;T extends Serializable &amp; CharSequence&gt;</code>.
-     * @noinspection HtmlTagCanBeJavadocTag
+     * The {@code &} symbol when used to extend a generic upper or lower bounds constrain
+     * or a type cast expression with an additional interface.
+     *
+     * <p>Generic type bounds extension:
+     * {@code class Comparable<T extends Serializable & CharSequence>}</p>
+     * <pre>
+     * CLASS_DEF
+     * |--MODIFIERS
+     * |--LITERAL_CLASS (class)
+     * |--IDENT (Comparable)
+     * +--TYPE_PARAMETERS
+     *     |--GENERIC_START (&lt;)
+     *     |--TYPE_PARAMETER
+     *     |   |--IDENT (T)
+     *     |   +--TYPE_UPPER_BOUNDS (extends)
+     *     |       |--IDENT (Serializable)
+     *     |       |--TYPE_EXTENSION_AND (&amp;)
+     *     |       +--IDENT (CharSequence)
+     *     +--GENERIC_END (&gt;)
+     * </pre>
+     *
+     * <p>Type cast extension:
+     * {@code return (CheckedFunction & Serializable) () -> Try.of(this::apply).get();}</p>
```

Comment:
Can we make a code as `return (CheckedFunction & Serializable) null;` ?
It is better to not have any extra code in sample but try to be compilable.

Reply:
done